charger/etrel.go

```diff
@@ -32,21 +32,21 @@ import (
 // https://github.com/RustyDust/sonnen-charger/blob/main/Etrel%20INCH%20SmartHome%20Modbus%20TCPRegisters.xlsx
 
 const (
-	// read
-	etrelRegChargeStatus = 0
-	etrelRegCurrents     = 14 // 16, 18
-	etrelRegPower        = 26
-	etrelRegSerial       = 990
-	etrelRegModel        = 1000
-	etrelRegHWVersion    = 1010
-	etrelRegSWVersion    = 1015
+	// input, read-only
+	etrelRegChargeStatus  = 0
+	etrelRegTargetCurrent = 4
+	etrelRegCurrents      = 14 // 16, 18
+	etrelRegPower         = 26
+	etrelRegSerial        = 990
+	etrelRegModel         = 1000
+	etrelRegHWVersion     = 1010
+	etrelRegSWVersion     = 1015
 	// etrelRegMaxPhaseCurrent  = 2
-	// etrelRegTargetCurrent    = 4 // power mgmt or modbus
-	// etrelRegSessionEnergy = 30
+	// etrelRegSessionEnergy    = 30
```

Comment:
@andig Weisst du warum wir hier nicht die SessionEnergy (und die ChargeTime) von der Box verwenden?
Ist da auch wieder irgendein Firmware-Bug drin?

Reply:
Wenn in der git history nix steht keine Ahnung.

Reply:
Ok, dann untersuche ich das nochmal.

Reply:
@premultiply falls das was hilft, hier die Ermittlung der Registerwerte im evcc-Modus `Min+PV` per `DUO_modbus_read.py`:

```
CONNECTOR 1 (24 registers)
--------------------------------------------------------------------------------
[0]Connector 1 Status: Charging
[1]Connector 1 Measured vehicle number of phases: Three phases
[2]Connector 1 EV max phase current: 15.74A
[4]Connector 1 Target current from power mgm or modbus: 6.00A
[6]Connector 1 Frequency: 49.99Hz
[8]Connector 1 L-N voltage (L1): 233.66V
[10]Connector 1 L-N voltage (L2): 235.80V
[12]Connector 1 L-N voltage (L3): 236.38V
[14]Connector 1 Current (L1): 6.03A
[16]Connector 1 Current (L2): 5.70A
[18]Connector 1 Current (L3): 5.87A
[20]Connector 1 Active power (L1): 1.36kW
[22]Connector 1 Active power (L2): 1.30kW
[24]Connector 1 Active power (L3): 1.33kW
[26]Connector 1 Active power (Total): 3.97kW
[28]Connector 1 Power factor: 0.963
[30]Connector 1 Total imported active energy in running session: 0.20kWh
[32]Connector 1 Running session duration (h:mm:ss): 0:15:30
[36]Connector 1 Running session departure time: 16-11-2022 16:00
[40]Connector 1 Running session ID: 0
[44]Connector 1 EV max power: 0.000kW
[46]Connector 1 EV required energy: 0.000kWh
[48]Connector 1 Meter value: 0.0kWh
[53]Connector 1 Charging mode: Unknown charging mode!
```

Reply:
`DUO_modbus_read.py` gibt es hier: https://etrelchargingsolutions.atlassian.net/wiki/spaces/Home/pages/2236121092/Modbus+Communication+with+Inch+products

Reply:
Kannst du das Gleiche mal bitte noch abfragen wenn das Fahrzeug angeschlossen und vorher etwas geladen hat und dann mittels evcc angehalten wurde (Modus Aus)?

Im Prinzip müsste man bei diesen Chargern auch eine Ladepunktauswahl für die Duo-Charger noch mit einbauen... 🤔

Reply:
Here we go:
während des Ladens im Modus `PV+Min`:
```
CONNECTOR 1 (24 registers)
--------------------------------------------------------------------------------
[0]Connector 1 Status: Charging
[1]Connector 1 Measured vehicle number of phases: Three phases
[2]Connector 1 EV max phase current: 9.41A
[4]Connector 1 Target current from power mgm or modbus: 6.00A
[6]Connector 1 Frequency: 49.95Hz
[8]Connector 1 L-N voltage (L1): 232.32V
[10]Connector 1 L-N voltage (L2): 235.21V
[12]Connector 1 L-N voltage (L3): 234.93V
[14]Connector 1 Current (L1): 6.06A
[16]Connector 1 Current (L2): 5.66A
[18]Connector 1 Current (L3): 5.90A
[20]Connector 1 Active power (L1): 1.36kW
[22]Connector 1 Active power (L2): 1.28kW
[24]Connector 1 Active power (L3): 1.31kW
[26]Connector 1 Active power (Total): 3.97kW
[28]Connector 1 Power factor: 0.960
[30]Connector 1 Total imported active energy in running session: 0.41kWh
[32]Connector 1 Running session duration (h:mm:ss): 0:06:16
[36]Connector 1 Running session departure time: 17-11-2022 15:00
[40]Connector 1 Running session ID: 0
[44]Connector 1 EV max power: 0.000kW
[46]Connector 1 EV required energy: 0.000kWh
[48]Connector 1 Meter value: 0.0kWh
[53]Connector 1 Charging mode: Unknown charging mode!
```
nach Modus `Aus` in evcc:
```
CONNECTOR 1 (24 registers)
--------------------------------------------------------------------------------
[0]Connector 1 Status: EVSE has paused charging
[1]Connector 1 Measured vehicle number of phases: Three phases
[2]Connector 1 EV max phase current: 9.41A
[4]Connector 1 Target current from power mgm or modbus: 0.00A
[6]Connector 1 Frequency: 49.96Hz
[8]Connector 1 L-N voltage (L1): 233.34V
[10]Connector 1 L-N voltage (L2): 235.74V
[12]Connector 1 L-N voltage (L3): 236.92V
[14]Connector 1 Current (L1): 0.08A
[16]Connector 1 Current (L2): 0.00A
[18]Connector 1 Current (L3): 0.00A
[20]Connector 1 Active power (L1): 0.00kW
[22]Connector 1 Active power (L2): 0.00kW
[24]Connector 1 Active power (L3): 0.00kW
[26]Connector 1 Active power (Total): 0.00kW
[28]Connector 1 Power factor: 0.000
[30]Connector 1 Total imported active energy in running session: 0.48kWh
[32]Connector 1 Running session duration (h:mm:ss): 0:07:23
[36]Connector 1 Running session departure time: 17-11-2022 15:00
[40]Connector 1 Running session ID: 0
[44]Connector 1 EV max power: 0.000kW
[46]Connector 1 EV required energy: 0.000kWh
[48]Connector 1 Meter value: 0.0kWh
[53]Connector 1 Charging mode: Unknown charging mode!
```

Reply:
Danke! Dann passt erstmal alles so wie erwartet. Annahmen bestätigt.
Die Sache mit den zwei Ladepunkten schaue ich mir aber später nochmal separat an.

Reply:
@andig Könnte man hier vielleicht auch via `connector` Parameter (Default 1) lösen wie z. B. in https://github.com/evcc-io/evcc/blob/master/charger/prachtalpha.go, oder?

Reply:
In der Doku steht nur Connector 1 drin- insofern hätte ich keine Ahnung welche Register für 2 gülten?

Reply:
Schau mal hier: https://etrelchargingsolutions.atlassian.net/wiki/download/attachments/820707329/Etrel%20INCH%20(Duo)%20SmartHome%20Modbus%20TCPRegisters.xlsx?api=v2

https://etrelchargingsolutions.atlassian.net/wiki/spaces/Home/pages/2236121092/Modbus+Communication+with+Inch+products

Reply:
Im Prinzip einfach ein Registeroffset von 100 (dezimal) für den zweiten LP

Reply:
Nice, ich hab die DUO nicht gesehen. Separater PR?

Reply:
Dokulinks sollten auch mit in den Quelltext, ich kannte die Etrel Seite gar nicht...

Reply:
Ja, neuer PR ist denke ich gut für connector